pmsoService/product_flow/features/deliverytrip/api/serializers/write.py

```
@@ -1,29 +1,186 @@
+    # Accept id (either DeliveryTripItems.id or product/material id) and optional item_kind
+    id = serializers.UUIDField(required=True)
+    sale_order_item_id = serializers.UUIDField(required=False, allow_null=True)
+    item_kind = serializers.ChoiceField(choices=["PRODUCT", "MATERIAL"], required=False)
+    quantity = serializers.IntegerField(min_value=1, required=True)
+
+    def validate(self, data):
+        has_sale = data.get("sale_order_item_id") is not None
+        if has_sale and not data.get("item_kind"):
+            raise serializers.ValidationError(
+                "'item_kind' is required when using 'sale_order_item_id'"
+            )
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)